pkg/util/error.go

```diff
@@ -84,3 +90,23 @@ func NestedErrorMessage(err error) string {
 
 	return res
 }
+
+// isRawCertRelatedError checks if the error message contains common
+// substrings related to certificate issues, such as expired or invalid certificates.
+// These errors may have types that are not easily identifiable (or not exported),
+// so we can't use errors.Is() to check for them.
+// In our code we try to detect these errors and replace them with a well-known error types like
+// ErrClientTLSCertExpired and ErrClientUnauthorized.
+// However, certain code paths may still return "raw" underlying errors and so
+// we need to handle these as well.
+func isRawCertRelatedError(err error) bool {
+	return IsUnauthorizedError(err) || IsTLSCertExpiredError(err)
+}
+
+func IsUnauthorizedError(src error) bool {
+	return strings.HasSuffix(strings.TrimRight(src.Error(), " \n"), msgUnauthorized)
+}
+
+func IsTLSCertExpiredError(src error) bool {
```

Comment:
Same here: Would be interesting to see some wrapped errors that will result in `true` here